recipes/arduinojson/all/conanfile.py

```diff
@@ -17,15 +19,20 @@ class ArduinojsonConan(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
     no_copy_source = True
 
+    def validate(self):
+        if Version(self.version) >= "6.21.0":
+            check_min_cppstd(self, 11)
+
     def package_id(self):
         self.info.clear()
 
     def layout(self):
         basic_layout(self, src_folder="src")
 
     def source(self):
+        has_arduinojson_root=Version(self.version) < "6.18.2"
         get(self, **self.conan_data["sources"][self.version],
-            destination=self.source_folder, strip_root=True)
+            destination=self.source_folder, strip_root=has_arduinojson_root)
```

Comment:
Could you please explain what error you have when not using stripped root?

Reply:
Did a quick local test to force it again, set to `True`. Version `6.18` builds, version `6.21.0` gives the following error:

~~~~
ERROR: arduinojson/6.21.0: Error in source() method, line 34
	get(self, **self.conan_data["sources"][self.version],
	ConanException: The zip file contains more than 1 folder in the root
~~~~

Reply:
Thank you for checking it, indeed it extracts to the root directly, so does not need to strip another root folder.